### PR TITLE
Add tntblast (thermonucleotideBLAST)

### PR DESCRIPTION
**Pull request by dfornika**
Describe your pull request here

Add tntblast (aka. [thermonucleotideBLAST](https://github.com/jgans/thermonucleotideBLAST))

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by dfornika** [comment]
I'm seeing an error:
```
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++  -O3 -Wall -fopenmp -std=c++14  -I.  -c tntblast.cpp
In file included from tntblast.h:6,
                 from tntblast.cpp:5:
sequence_data.h:12:10: fatal error: zlib.h: No such file or directory
   12 | #include <zlib.h>
      |          ^~~~~~~~
```

...but I've included `zlib` under the `build` requirements. I'm not sure why the compiler can't find it. Maybe I need to add `-lz` to the compiler flags?

**Comment by dfornika** [comment]
That (using `CXX`) does seem to have helped. The compiler step is completing now.

I'm now seeing:

```
21:19:18 BIOCONDA INFO (OUT) Packaging tntblast
21:19:18 BIOCONDA INFO (OUT) INFO:conda_build.build:Packaging tntblast
21:19:18 BIOCONDA INFO (OUT) Packaging tntblast-2.61-hdcf5f25_0
21:19:18 BIOCONDA INFO (OUT) INFO:conda_build.build:Packaging tntblast-2.61-hdcf5f25_0
21:19:30 BIOCONDA INFO (OUT) shutil.move(work)=/opt/conda/conda-bld/tntblast_1705439682948/work, dest=/opt/conda/conda-bld/tntblast_1705439682948/work_moved_tntblast-2.61-hdcf5f25_0_linux-64)
21:19:30 BIOCONDA INFO (OUT) INFO:conda_build.utils:shutil.move(work)=/opt/conda/conda-bld/tntblast_1705439682948/work, dest=/opt/conda/conda-bld/tntblast_1705439682948/work_moved_tntblast-2.61-hdcf5f25_0_linux-64)
21:19:30 BIOCONDA INFO (OUT) Reloading output folder: /opt/conda/conda-bld
21:19:30 BIOCONDA INFO (OUT) 25l[+] 0.0s
21:19:30 BIOCONDA INFO (OUT) opt/conda/conda-bld/linux-64                       717.0 B @  16.7MB/s  0.0s
21:19:30 BIOCONDA INFO (OUT) opt/conda/conda-bld/noarch                         127.0 B @   5.1MB/s  0.0s
21:19:38 BIOCONDA INFO (OUT) 25h
21:19:38 BIOCONDA INFO (OUT) ## Package Plan ##
21:19:38 BIOCONDA INFO (OUT) 
21:19:38 BIOCONDA INFO (OUT)   environment location: /opt/conda/conda-bld/tntblast_1705439682948/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
21:19:38 BIOCONDA INFO (OUT) 
21:19:38 BIOCONDA INFO (OUT) 
21:19:38 BIOCONDA INFO (OUT) The following NEW packages will be INSTALLED:
21:19:38 BIOCONDA INFO (OUT) 
21:19:38 BIOCONDA INFO (OUT)     _libgcc_mutex: 0.1-conda_forge   conda-forge
21:19:38 BIOCONDA INFO (OUT)     _openmp_mutex: 4.5-2_gnu         conda-forge
21:19:38 BIOCONDA INFO (OUT)     libgcc-ng:     13.2.0-h807b86a_3 conda-forge
21:19:38 BIOCONDA INFO (OUT)     libgomp:       13.2.0-h807b86a_3 conda-forge
21:19:38 BIOCONDA INFO (OUT)     libstdcxx-ng:  13.2.0-h7e041cc_3 conda-forge
21:19:38 BIOCONDA INFO (OUT)     libzlib:       1.2.13-hd590300_5 conda-forge
21:19:38 BIOCONDA INFO (OUT)     llvm-openmp:   8.0.1-hc9558a2_0  conda-forge
21:19:38 BIOCONDA INFO (OUT)     openmp:        8.0.1-0           conda-forge
21:19:38 BIOCONDA INFO (OUT)     tntblast:      2.61-hdcf5f25_0   local
21:19:38 BIOCONDA INFO (OUT)     zlib:          1.2.13-hd590300_5 conda-forge
21:19:38 BIOCONDA INFO (OUT) 
21:19:38 BIOCONDA INFO (OUT) Preparing transaction: ...working... done
21:19:38 BIOCONDA INFO (OUT) Verifying transaction: ...working... done
21:19:38 BIOCONDA INFO (OUT) Executing transaction: ...working... done
21:19:38 BIOCONDA INFO (OUT) export PREFIX=/opt/conda/conda-bld/tntblast_1705439682948/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
21:19:38 BIOCONDA INFO (OUT) export SRC_DIR=/opt/conda/conda-bld/tntblast_1705439682948/test_tmp
21:19:39 BIOCONDA INFO (OUT) + tntblast
21:19:39 BIOCONDA INFO (OUT) + grep v.2.61
21:19:41 BIOCONDA INFO (OUT) Tests failed for tntblast-2.61-hdcf5f25_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
21:19:41 BIOCONDA INFO (OUT) WARNING:conda_build.build:Tests failed for tntblast-2.61-hdcf5f25_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
21:19:41 BIOCONDA INFO (OUT) TESTS FAILED: tntblast-2.61-hdcf5f25_0.tar.bz2
21:19:51 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmp59ruttm8/build_script.bash:/opt/build_script.bash -v /opt/mambaforge/envs/bioconda/conda-bld/:/opt/host-conda-bld -v /home/vsts/work/1/s/recipes/tntblast:/opt/recipe -e LANG=C.UTF-8 -e HOST_USER_ID=1001 quay.io/bioconda/bioconda-utils-build-env-cos7:2.11.1 /bin/bash /opt/build_script.bash

```

**Comment by dfornika** [comment]
The linter doesn't seem to like the syntax I'm using for the test. This test:

```
{ tntblast 2>&1 || true; } | grep "v.{{ version }}"
```

...results in this linter error:

```
ruamel.yaml.scanner.ScannerError: while scanning a block scalar
  in "<unicode string>", line 35, column 34:
        - { tntblast 2>&1 || true; } | grep "v.2.61"
                                     ^ (line: 35)
expected a comment or a line break, but found 'g'
  in "<unicode string>", line 35, column 36:
        - { tntblast 2>&1 || true; } | grep "v.2.61"
                                       ^ (line: 35)
```

File: recipes/tntblast/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+sed -i.bak 's/-DUSE_MPI//g' Makefile
+sed -i.bak '/BLAST_DIR =/d' Makefile
+
+export INCLUDE_PATH="${BUILD_PREFIX}/include"
+export LIBRARY_PATH="${BUILD_PREFIX}/lib"
+export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${BUILD_PREFIX}/lib"
+
+export LDFLAGS="-lm -lz -L${BUILD_PREFIX}/lib"
+export CPPFLAGS="-I${BUILD_PREFIX}/include"
```
Comment (martin-g) [inline recipes/tntblast/build.sh:11]: You could make use of the earlier declared `$INCLUDE_PATH` and `$LIBRARY_PATH`

**Comment by dfornika** [comment]
I can see from the build logs that the test is failing, but I'm not sure why they're failing and I don't know what options I have for debugging/troubleshooting.

This is what I'm seeing in the build log:

```
23:01:58 BIOCONDA INFO (OUT) TEST START: /opt/conda/conda-bld/linux-64/tntblast-2.61-hdcf5f25_0.tar.bz2
23:01:58 BIOCONDA INFO (OUT) Renaming work directory '/opt/conda/conda-bld/tntblast_1705445757825/work' to '/opt/conda/conda-bld/tntblast_1705445757825/work_moved_tntblast-2.61-hdcf5f25_0_linux-64'
23:01:58 BIOCONDA INFO (OUT) INFO:conda_build.utils:Renaming work directory '/opt/conda/conda-bld/tntblast_1705445757825/work' to '/opt/conda/conda-bld/tntblast_1705445757825/work_moved_tntblast-2.61-hdcf5f25_0_linux-64'
23:01:58 BIOCONDA INFO (OUT) shutil.move(work)=/opt/conda/conda-bld/tntblast_1705445757825/work, dest=/opt/conda/conda-bld/tntblast_1705445757825/work_moved_tntblast-2.61-hdcf5f25_0_linux-64)
23:01:58 BIOCONDA INFO (OUT) opt/conda/conda-bld/noarch                         127.0 B @   3.4MB/s  0.0s
23:02:07 BIOCONDA INFO (OUT) 25h
23:02:07 BIOCONDA INFO (OUT) ## Package Plan ##
23:02:07 BIOCONDA INFO (OUT) 
23:02:07 BIOCONDA INFO (OUT)   environment location: /opt/conda/conda-bld/tntblast_1705445757825/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
23:02:07 BIOCONDA INFO (OUT) 
23:02:07 BIOCONDA INFO (OUT) 
23:02:07 BIOCONDA INFO (OUT) The following NEW packages will be INSTALLED:
23:02:07 BIOCONDA INFO (OUT) 
23:02:07 BIOCONDA INFO (OUT)     _libgcc_mutex: 0.1-conda_forge   conda-forge
23:02:07 BIOCONDA INFO (OUT)     _openmp_mutex: 4.5-2_gnu         conda-forge
23:02:07 BIOCONDA INFO (OUT)     libgcc-ng:     13.2.0-h807b86a_3 conda-forge
23:02:07 BIOCONDA INFO (OUT)     libgomp:       13.2.0-h807b86a_3 conda-forge
23:02:07 BIOCONDA INFO (OUT)     libstdcxx-ng:  13.2.0-h7e041cc_3 conda-forge
23:02:07 BIOCONDA INFO (OUT)     libzlib:       1.2.13-hd590300_5 conda-forge
23:02:07 BIOCONDA INFO (OUT)     llvm-openmp:   8.0.1-hc9558a2_0  conda-forge
23:02:07 BIOCONDA INFO (OUT)     openmp:        8.0.1-0           conda-forge
23:02:07 BIOCONDA INFO (OUT)     tntblast:      2.61-hdcf5f25_0   local
23:02:07 BIOCONDA INFO (OUT)     zlib:          1.2.13-hd590300_5 conda-forge
23:02:07 BIOCONDA INFO (OUT) 
23:02:07 BIOCONDA INFO (OUT) Preparing transaction: ...working... done
23:02:08 BIOCONDA INFO (OUT) Verifying transaction: ...working... done
23:02:08 BIOCONDA INFO (OUT) Executing transaction: ...working... done
23:02:08 BIOCONDA INFO (OUT) export PREFIX=/opt/conda/conda-bld/tntblast_1705445757825/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
23:02:08 BIOCONDA INFO (OUT) export SRC_DIR=/opt/conda/conda-bld/tntblast_1705445757825/test_tmp
23:02:09 BIOCONDA INFO (OUT) + tntblast
23:02:09 BIOCONDA INFO (OUT) + grep v.2.61
23:02:09 BIOCONDA INFO (OUT) + true
23:02:10 BIOCONDA INFO (OUT) Tests failed for tntblast-2.61-hdcf5f25_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
23:02:10 BIOCONDA INFO (OUT) WARNING:conda_build.build:Tests failed for tntblast-2.61-hdcf5f25_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
23:02:10 BIOCONDA INFO (OUT) TESTS FAILED: tntblast-2.61-hdcf5f25_0.tar.bz2
23:02:24 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmprxpbbghe/build_script.bash:/opt/build_script.bash -v /opt/mambaforge/envs/bioconda/conda-bld/:/opt/host-conda-bld -v /home/vsts/work/1/s/recipes/tntblast:/opt/recipe -e LANG=C.UTF-8 -e HOST_USER_ID=1001 quay.io/bioconda/bioconda-utils-build-env-cos7:2.11.1 /bin/bash /opt/build_script.bash
```

**Comment by dfornika** [comment]
This may be a clue:

```
17:17:08 BIOCONDA INFO (OUT) export PREFIX=/opt/conda/conda-bld/tntblast_1705511527334/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
17:17:08 BIOCONDA INFO (OUT) export SRC_DIR=/opt/conda/conda-bld/tntblast_1705511527334/test_tmp
17:17:09 BIOCONDA INFO (OUT) + tntblast
17:17:09 BIOCONDA INFO (OUT) $SRC_DIR/run_test.sh: line 7: tntblast: command not found
17:17:09 BIOCONDA INFO (OUT) + true
17:17:09 BIOCONDA INFO (OUT) + grep v.2.61
17:17:09 BIOCONDA INFO (OUT) + tntblast
17:17:09 BIOCONDA INFO (OUT) + true
17:17:11 BIOCONDA INFO (OUT) Tests failed for tntblast-2.61-hdcf5f25_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
```

Note:

```
$SRC_DIR/run_test.sh: line 7: tntblast: command not found
```

So it appears that the `tntblast` executable isn't making it on to the `PATH` when tests are being run.

**Comment by dfornika** [comment]
Yay, we're passing on linux.

On OSX it looks like there's a conflict because we're asking for both `openmp` and `llvm-openmp`, so we need to limit `openmp` to linux only.

```
18:33:16 BIOCONDA INFO (ERR) The reported errors are:
18:33:16 BIOCONDA INFO (ERR) - Encountered problems while solving:
18:33:16 BIOCONDA INFO (ERR) -   - package llvm-openmp-15.0.4-h61d9ccf_0 has constraint openmp 15.0.4|15.0.4.* conflicting with openmp-4.0.0-0
```

**Comment by dfornika** [comment]
@BiocondaBot please add label

**Review by thanhleviet — APPROVED** [review]
yay, LGTM

**Comment by dfornika** [comment]
@BiocondaBot please merge

**Pull request merged**